main.go

```diff
@@ -279,7 +330,10 @@ func createMapping(cfg elasticsearchConfig) error {
 		req.SetBasicAuth(cfg.user, cfg.pass)
 	}
 	req.Header.Set("Content-Type", "application/json")
-	resp, err := http.DefaultClient.Do(req)
+
+	req.WithContext(ctx)
```

Comment:
```suggestion
	req = req.WithContext(ctx)
```

WithContext does not mutate the original context.